#18 combine name and email display

florianm opened the pull request:
As discussed in (messy and closed) PR #22 this PR combines the display of name/email combo fields by hiding the name field, and letting the email field render the email address (if given) with the specified name field (if given).

wardi commented:
we still need the thing that checks for `'display_field': False`, right?  Alternatively we could use `'display_snippet': null`.

florianm commented:
Strangely, `display_field: false` just works - although I can't find where that happens, at least there's nothing acting upon `display_field` or manipulating exclude fields in ckan or scheming as far as I can grep. Any ideas?

wardi commented:
Hmm. doesn't seem to hide the fields when I try it.

florianm commented:
Found why it works on my end - I'm still using my much too clever [additional info display snippet](https://github.com/florianm/ckanext-scheming/blob/master/ckanext/scheming/templates/scheming/package/snippets/additional_info.html) with special treatment for author and maintainer fields.

wardi commented:
@florianm ok, should we close this too, or will you add the display_snippet is None check to this PR?

florianm commented:
@wardi sorry I got sidetracked - I will add the check to this PR today.

florianm commented:
@wardi added a property "hide_field", formatted additional_info.html for readability and added check for hide_field there, added email example to camel_photos.json.

Up for discussion:

* The author-hiding property is named `hide_field` (instead of `display_field`) so the template can check for its existence and whether to hide the field in one operation.
* I've added values for the email example to tests which would otherwise fail. I can create a dataset without author and email fields through the GUI, but the tests will fail on blank author/email.

florianm commented:
@wardi I need to double-check whether dataset notes are hidden inadvertently, sorry that's one thing I might have clobbered last minute!

florianm commented:
@wardi there's travis errors with the solr server that I'm sure I didn't touch. Any ideas whether that's a problem to do with my changed templates?
Also, please allow me a few days of thinking (my debugging appears to slow down when nappie changing instead of sleeping) to figure out why dataset notes are not showing on my end.
